examples/nextjs/server.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const next = require('next');
+
+const port = parseInt(process.env.PORT, 10) || 3000;
+const dev = process.env.NODE_ENV !== 'production';
+const app = next({ dev });
+const handle = app.getRequestHandler();
+const createRouteMiddleware = require('@paralleldrive/react-feature-toggles/dist/create-route-middleware');
+const features = require('./features');
+
+const featureToggleHandler = createRouteMiddleware(features);
+
+app.prepare().then(() => {
+  const server = express();
+
+  // This will check the feature and set the correct status code for this route.
+  // 200 if the feature is enabled, otherwise it will be 404.
+  server.use('/profile', featureToggleHandler('profile'));
```

Comment:
Both route handlers could be consolidated. A single handler can delegate to middleware, and we could create a function which takes the feature name and the route handler and returns a function that applies both with a single app.use statement:

```js
server.use('/profile', createHandler({
  requiresFeature: 'profile',
  get: (req, res) => {
    return app.render(req, res, '/profile', req.query)
  }
));
```

Reply:
@ericelliott, would it be better to use server.get instead of server.use?

I am thinking that if we apply server.use to the resulting handler function, then the resulting handler would prematurely render before the remaining middleware has been applied to the req, res objects. 

So maybe this? 

```javascript
server.get('/profile', createHandler({
  requiresFeature: 'profile',
  handler: (req, res) => {
    return app.render(req, res, '/profile', req.query)
  }
));
```

But I could be missing something, a reason why you used server.use on purpose?

Reply:
Nvm. I think I see why you chose to use `server.use`, it allows you to set all the method handlers in a single function if you want which can handle more cases like api calls etc.

```javascript
server.use('/profile', createHandler({
  requiresFeature: 'profile',
  get: (req, res) => {
    return app.render(req, res, '/profile', req.query)
  },
  post: (req, res) => {
    // something
  },
  delete: (req, res) => {
    // something
  },
  put: (req, res) => {
    // something
  },
));
```

Reply:
I will do my best to have this change ready for review tomorrow.